src/test/java/uk/gov/hmcts/darts/authorisation/controller/impl/AuthorisationControllerTest.java

```diff
@@ -54,25 +53,24 @@ void getUserStateOk() {
 
         UserState userState = UserState.builder()
             .userId(123)
-            .userName("UserName")
+            .userName("testUsername")
             .roles(newRoles)
             .isActive(true)
             .build();
-        when(authorisationApi.getAuthorisation(userAccountEntity.getEmailAddress())).thenReturn(Optional.of(userState));
+        when(authorisationApi.getAuthorisation(userAccountEntity.getId())).thenReturn(userState);
 
         UserState userStateResponse = controller.getUserState();
 
         assertNotNull(userStateResponse);
-        assertEquals("UserName", userStateResponse.getUserName());
+        assertEquals("testUsername", userStateResponse.getUserName());
         assertTrue(userStateResponse.getIsActive());
     }
 
     @Test
     void getUserStateForbiddenWhenNoEmailAddress() {
-        UserAccountEntity userAccountEntity = CommonTestDataUtil.createUserAccount();
-        when(authorisationApi.getCurrentUser()).thenReturn(userAccountEntity);
-        when(authorisationApi.getAuthorisation(userAccountEntity.getEmailAddress())).thenReturn(Optional.empty());
+        when(authorisationApi.getCurrentUser()).thenThrow(new DartsApiException(AuthorisationError.USER_DETAILS_INVALID));
         var exception = assertThrows(DartsApiException.class, () -> controller.getUserState());
-        assertEquals(AuthorisationError.USER_NOT_AUTHORISED_FOR_ENDPOINT, exception.getError());
+        assertEquals(AuthorisationError.USER_DETAILS_INVALID, exception.getError());
```

Comment:
Author note: These mocks set up a scenario that wasn't possible, so correcting here.